test/functorch/test_control_flow.py

```diff
@@ -2715,5 +2767,108 @@ def forward(self, l_inp_, l_tmp_):
 
 instantiate_parametrized_tests(TestControlFlowTraced)
 
+class DynamicCondModel(torch.nn.Module):
+    def __init__(self, input_size=16, hidden_size=64, output_size=10):
+        super(DynamicCondModel, self).__init__()
+        self.fc1_0 = torch.nn.Linear(input_size, hidden_size)
+        self.fc1_1 = torch.nn.Linear(input_size, 32)
+        self.fc1_2 = torch.nn.Linear(32, hidden_size)
+        self.relu = torch.nn.ReLU()
+        self.fc2 = torch.nn.Linear(hidden_size, output_size)
+
+    def forward(self, x):
+        def true_fn(x):
+            return self.fc1_0(x)
+        
+        def false_fn(x):
+            x = self.fc1_1(x)
+            return self.fc1_2(x)
+        
+        # use PyTorch control flow API
+        pred = torch.tensor(x.sum() > 0, device="cuda")
+        x = cond(pred, true_fn, false_fn, [x])
+        
+        x = self.relu(x)
+        x = self.fc2(x)
+
+        return x
+
+class DynamicWhileNestedModel(torch.nn.Module):
+    def __init__(self, out_iter=5):
+        super(DynamicWhileNestedModel, self).__init__()
+        self.fc = torch.nn.Linear(2, 2)
+        self.relu = torch.nn.ReLU()
+        self.out_iter = out_iter
+        self.flag = torch.tensor(True, dtype=torch.bool, device="cuda")
+        self.idx = torch.zeros(1, dtype=torch.int64, device="cuda")
+        self.results = torch.zeros((5, 2, 2), device="cuda")
+
+    def forward(self, x, it):
+        def outer_cond_fn(it, x, idx):
+            return it.sum() < self.out_iter
+
+        def outer_body_fn(it, x, idx):
+            self.flag.fill_(True)
+            
+            def cond_fn(x, idx):
+                return torch.logical_or(x.sum() < 0, self.flag)
+                # return self.flag -> I doubt this a bug...
+                # this causes an issue in the "cudagraphs" backend
+                # error msg: torch._dynamo.exc.BackendCompilerFailed: backend='cudagraphs' raised:
+                # UnsupportedAliasMutationException: torch.while_loop's body_fn might be modifying the input!
```

Comment:
The error message is a little bit misleading, if we return self.flag, it's actually an input aliasing, which we haven't yet supported right now.